service/application_test.go

```diff
@@ -348,3 +353,68 @@ service:
 	assert.NoError(t, err)
 	return cfg
 }
+
+func TestApplication_updateActiveConfig(t *testing.T) {
+	factories, err := defaultcomponents.Components()
+	require.NoError(t, err)
+	ctx := context.Background()
+	sentinelError := errors.New("sentinel error")
+	returnConfigFactoryFn := func(cfg *configmodels.Config, err error) ConfigFactory {
+		return func(*cobra.Command, component.Factories) (*configmodels.Config, error) {
+			return cfg, err
+		}
+	}
+
+	tests := []struct {
+		name          string
+		configFactory ConfigFactory
+		service       *service
+	}{
+		{
+			name:          "first_load_err",
+			configFactory: returnConfigFactoryFn(nil, sentinelError),
+		},
+		{
+			name:          "retire_service_ok_load_err",
+			configFactory: returnConfigFactoryFn(nil, sentinelError),
+			service: &service{
+				logger:          zap.NewNop(),
+				builtExporters:  builder.Exporters{},
+				builtPipelines:  builder.BuiltPipelines{},
+				builtReceivers:  builder.Receivers{},
+				builtExtensions: builder.Extensions{},
+			},
+		},
+		{
+			name:          "retire_service_ok_load_ok",
+			configFactory: returnConfigFactoryFn(constructMimumalOpConfig(t, factories), nil),
+			service: &service{
+				logger:          zap.NewNop(),
+				builtExporters:  builder.Exporters{},
+				builtPipelines:  builder.BuiltPipelines{},
+				builtReceivers:  builder.Receivers{},
+				builtExtensions: builder.Extensions{},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			app := Application{
+				logger:        zap.NewNop(),
+				configFactory: tt.configFactory,
+				factories:     factories,
+				service:       tt.service,
+			}
+
+			err := app.updateActiveConfig(ctx)
```

Comment:
Is it valid to call this before Application.Run is called? Should it be valid? Can we document it?

Reply:
It requires the `app.factories` and the `app.configFactory`, documented it.